packages/api/amplify_api/lib/src/graphql/graphql_send_request.dart

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'dart:convert';
+
+import 'package:amplify_core/amplify_core.dart';
+import 'package:http/http.dart' as http;
+import 'package:meta/meta.dart';
+
+import '../util.dart';
+import 'graphql_response_decoder.dart';
+
+@internal
+Future<GraphQLResponse<T>> graphQLSendRequest<T>({
+  required GraphQLRequest<T> request,
+  required http.Client client,
+  required Uri uri,
+}) async {
+  final body = {'variables': request.variables, 'query': request.document};
+  final graphQLResponse = await client.post(uri, body: json.encode(body));
+
+  final responseBody = json.decode(graphQLResponse.body);
+
+  if (responseBody is! Map<String, dynamic>) {
+    throw Exception('Response from GraphQL client incorrect type.');
```

Comment:
nit: Could this be `ApiException` and change the message to something like "unable to parse GraphQLResponse from server response which was not a JSON object." Just thinking it might be more helpful to folks than something about type errors.

Reply:
Also, for the underlying error argument to `ApiException`, would it make sense to include the status code and the response body? If someone had an error here, they might have no way of seeing the server error unless we surface along with this error.

Reply:
It might also make sense to wrap the `json.decode(` in a try/catch as I would expect that to throw an error if the body wasn't JSON-able.